extensions/amp-social-share/1.0/amp-social-share.js

```diff
@@ -71,24 +79,72 @@ const systemShareSupported = (viewer, platform) => {
   return 'share' in navigator && !isChromeWebview;
 };
 
+/**
+ * @private
+ * @param {!Element} element
+ * @param {Array<MutationRecord>} mutations
+ * @return {!JsonObject|undefined}
+ */
+const updateTypeConfig = (element, mutations) => {
+  let typeOldValue;
+  let mutatedEligibleAttribute;
+
+  // Check all mutations since we want to catch any 'data-param-*' attributes
+  mutations.forEach((mutation) => {
+    if (
+      ['type', 'data-target', 'data-share-endpoint'].includes(
+        mutation.attributeName
+      ) ||
+      (mutation.attributeName && mutation.attributeName.includes('data-param-'))
+    ) {
+      mutatedEligibleAttribute = true;
+      typeOldValue = mutation.attributeName === 'type' && mutation.oldValue;
+    }
+  });
+
+  // If no matching attribute changes exit and do nothing
+  if (!mutatedEligibleAttribute) {
+    return;
+  }
+
+  // If 'type' attribute was changed, remove the class of the old 'type'
+  if (typeOldValue) {
+    element.classList.remove(`amp-social-share-${typeOldValue}`);
+  }
+
+  const typeConfig = getTypeConfigOrUndefined(element);
+  if (!typeConfig) {
+    toggle(element, false);
+    return;
+  }
+  element.classList.add(`amp-social-share-${element.getAttribute('type')}`);
+  return typeConfig;
+};
+
 class AmpSocialShare extends PreactBaseElement {
   /** @override */
   init() {
-    const viewer = Services.viewerForDoc(this.element);
-    const platform = Services.platformFor(window);
-    const type = userAssert(
-      this.element.getAttribute('type'),
-      'The type attribute is required. %s',
-      this.element
-    );
-    const typeConfig = getTypeConfigOrUndefined(type, viewer, platform);
+    const typeConfig = getTypeConfigOrUndefined(this.element);
     // Hide/ignore component if typeConfig is undefined
     if (!typeConfig) {
       toggle(this.element, false);
       return;
     }
+    this.element.classList.add(
+      `amp-social-share-${this.element.getAttribute('type')}`
+    );
+
+    const mu = new MutationObserver((mutations) => {
```

Comment:
There's the `mutationObserverCallback` - we should be able to override that instead of creating a new observer.

Reply:
Updated, had to make a few updates to the `base-element` class.  Let me know your thoughts!